profiles/views_test.py

```diff
@@ -0,0 +1,136 @@
+"""
+Tests for profile view
+"""
+import json
+
+from dateutil.parser import parse
+from django.contrib.auth.models import User
+from django.core.urlresolvers import reverse
+from django.test import TestCase
+from rest_framework.status import (
+    HTTP_403_FORBIDDEN,
+    HTTP_405_METHOD_NOT_ALLOWED,
+)
+
+from profiles.factories import ProfileFactory
+from profiles.models import Profile
+from profiles.serializers import (
+    ProfileLimitedSerializer,
+    ProfilePrivateSerializer,
+    ProfileSerializer,
+)
+
+
+class ProfileTests(TestCase):
+    """
+    Tests for GET on profile view
+    """
+
+    def setUp(self):
+        """
+        Create a user and profile
+        """
+        super(ProfileTests, self).setUp()
+
+        self.user1_credentials = {
+            "username": "user1",
+            "password": "pass"
+        }
+        self.user1 = User.objects.create_user(**self.user1_credentials)
+        self.url1 = reverse('profile-detail', kwargs={'user': self.user1.username})
+
+        self.user2_credentials = {
+            "username": "user2",
+            "password": "pass"
+        }
+        self.user2 = User.objects.create_user(**self.user2_credentials)
+        self.url2 = reverse('profile-detail', kwargs={'user': self.user2.username})
+
+    def test_no_anonymous_access(self):
```

Comment:
We also need to test that we called `self.check_object_permissions` somewhere, otherwise the `has_object_permission` function won't be called. One reason I like integration tests is that they detect these edge cases automatically. But I think in this case we should be ok to break it up and assert that `has_object_permission` was called